content/kubernetes/re-clusters/redis-on-flash.md

```diff
@@ -0,0 +1,83 @@
+    /kubernetes/re-clusters/redis-on-flash/,
+]  
+---
+
+## Prerequisites
+
+Redis Enterprise Software for Kubernetes supports using NVMe (nonvolatile memory express) SSDs (solid state drives) for flash storage with the Redis on Flash feature.
+
+Before creating your Redis clusters or databases, these SSDs must be:
+
+- [locally attached to worker nodes in your Kubernetes cluster](https://kubernetes.io/docs/concepts/storage/volumes/#local)
+- [formatted and mounted]() on the worker nodes that will run Redis Enterprise pods
+- [provisioned as local persistent volumes](https://kubernetes.io/docs/concepts/storage/volumes/#local)
+  - You can use a [local volume provisioner](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/master/README.md) to do this [dynamically](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#dynamic)
+- a [StorageClass](https://kubernetes.io/docs/concepts/storage/storage-classes/#local) resource specifying a unique name
+
+{{< note >}}
+Redis on Flash support is currently in preview. To enable this feature, create a boolean environment variable with the name `ENABLE_ALPHA_FEATURES` to `True` in either the `redis-enterprise-operator` pod spec or the `operator-environment-config` ConfigMap.
+{{< /note >}}
+
+## Create Redis Enterprise cluster
+
+To deploy a Redis Enterprise cluster (REC) with flash storage, you'll need to specify the following in the `redisOnFlashSpec` section of your [REC custom resource]:
+
+- enable Redis on Flash (`enabled: true`)
+- flash storage driver (`flashStorageEngine`)
+  - The only currently supported value is `rocksdb`
+- storage class name (`storageClassName`)
+- minimal flash disk size (`flashDiskSize`)
+
+Here is an example of an REC custom resource with these attributes:
+
+```YAML
+apiVersion: app.redislabs.com/v1
+kind: RedisEnterpriseCluster
+metadata:
+  name: "rec"
+spec:
+  
+  nodes: 3
+  redisOnFlashSpec:
+    enabled: true
+    flashStorageEngine: rocksdb
+    storageClassName: local-scsi
+    flashDiskSize: 100G
+```
+
+### Create Redis Enterprise database
+
+By default, any new database will use RAM only. To create a Redis Enterprise database (REDB) that can use flash storage, specify the following in the `redisEnterpriseCluster` section of the REDB custom resource definition:
+
+- `isRof: true` enables Redis on Flash
+- `rofRamSize` defines the RAM capacity for the database
+
+Below is an example REDB custom resource:
+
+```YAML
+apiVersion: app.redislabs.com/v1alpha1
+kind: RedisEnterpriseDatabase
+metadata:
+  name: rof-redb
+spec:
+  redisEnterpriseCluster:
+    name: rec
+  isRof: true
+  memorySize: 2GB
+  rofRamSize: 0.5GB
+```
+
+{{< note >}}
+This example defines both `memorySize` and `rofRamSize`. When using Redis on Flash, `memorySize` refers to the combined memory size (RAM + flash), while `rofRamSize` specifies only the RAM capacity for the database. `rofRamSize` must be at least 10% of `memorySize`.
```

Comment:
This seems counterintuitive, so just want to double check: rofSize refers to the amount of RAM and not the amount of flash?

Reply:
That's my understanding but I will check with @heinrich-redislabs to clarify.

Reply:
Indeed it sounds counter intuitive, let me clarify:
memorySize always refers to the combined memory allocated for the database.
In case on non-RoF database, it's simply the RAM size. In case of RoF database, it's combined RAM+Flash size.
rofRamSize - refers to the RAM portion of an RoF database.